lnwallet/wallet.go

```diff
@@ -1287,6 +1291,34 @@ func (l *LightningWallet) initOurContribution(reservation *ChannelReservation,
 	reservation.partialState.RevocationProducer = producer
 	reservation.ourContribution.ChannelConstraints = l.Cfg.DefaultConstraints
 
+	// If taproot channels are active, then we'll generate our verification
+	// nonce here. We'll use this nonce to verify the signature for our
+	// local commitment transaction. If we need to force close, then this
+	// is also what'll be used to sign that transaction.
+	if reservation.partialState.ChanType.IsTaproot() {
+		firstNoncePreimage, err := taprootNonceProducer.AtIndex(0)
```

Comment:
So this is the nonce that's sent in `open_channel` by the funder and can be used to generate the final sig when receiving `funding_signed.nonce` from the fundee right?

Reply:
Correct, this is where we use the counter based method to generate nonces we can always arrive at again. 

In part 4, I started to also bind the nonce creation with the tx hash of the tx we're about to sign. However at this point, we need to send a nonce, and don't yet know the tx hash so we can't do so. 

We'll still do that each time we go to sign a new commitment, as the tx hash then is already known.